velox/functions/prestosql/DecimalArithmetic.cpp

```diff
@@ -97,6 +98,52 @@ class DecimalBaseFunction : public exec::VectorFunction {
   const uint8_t bRescale_;
 };
 
+template <
+    typename R /* Result Type */,
+    typename A /* Argument */,
+    typename Operation /* Arithmetic operation */>
+class DecimalUnaryBaseFunction : public exec::VectorFunction {
```

Comment:
You probably also want to overwrite `supportsFlatNoNullsFastPath()`

Reply:
Done.